main.go

```diff
@@ -0,0 +1,186 @@
+)
+
+// Config ...
+type Config struct {
+	RepositoryURL string `env:"repository_url,required"`
+
+	GithubAPIToken   stepconf.Secret `env:"github_api_token"`
+	GithubHost       string          `env:"github_host"`
+	GithubAPIBaseURL string          `env:"github_api_base_url"`
+
+	GitlabAPIToken   stepconf.Secret `env:"gitlab_api_token"`
+	GitlabHost       string          `env:"gitlab_host"`
+	GitlabAPIBaseURL string          `env:"gitlab_api_base_url"`
+}
+
+func main() {
+	var cfg Config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+
+	// Fix repository URL
+	cfg.RepositoryURL = strings.TrimLeft(cfg.RepositoryURL, "https://")
+
+	stepconf.Print(cfg)
+	fmt.Println()
+
+	validateInputs(cfg)
+
+	//
+	// Set local envs for the step
+	{
+		if err := os.Setenv("GIT_REPOSITORY_URL", cfg.RepositoryURL); err != nil {
+			failf("Failed to set env GIT_REPOSITORY_URL, error: %s", err)
+		}
+
+		// Github
+		{
+			if string(cfg.GithubAPIToken) != "" {
+				if err := os.Setenv("DANGER_GITHUB_API_TOKEN", string(cfg.GithubAPIToken)); err != nil {
+					failf("Failed to set env DANGER_GITHUB_API_TOKEN, error: %s", err)
+				}
+			}
+
+			if string(cfg.GithubHost) != "" {
+				if err := os.Setenv("DANGER_GITHUB_HOST", string(cfg.GithubHost)); err != nil {
+					failf("Failed to set env DANGER_GITHUB_HOST, error: %s", err)
+				}
+			}
+
+			if string(cfg.GithubAPIBaseURL) != "" {
+				if err := os.Setenv("DANGER_GITHUB_API_BASE_URL", string(cfg.GithubAPIBaseURL)); err != nil {
+					failf("Failed to set env DANGER_GITHUB_API_BASE_URL, error: %s", err)
+				}
+			}
+		}
+
+		// Gitlab
+		{
+			if string(cfg.GitlabAPIToken) != "" {
+				if err := os.Setenv("DANGER_GITLAB_API_TOKEN", string(cfg.GitlabAPIToken)); err != nil {
+					failf("Failed to set env DANGER_GITLAB_API_TOKEN, error: %s", err)
+				}
+			}
+
+			if string(cfg.GitlabHost) != "" {
+				if err := os.Setenv("DANGER_GITLAB_HOST", string(cfg.GitlabHost)); err != nil {
+					failf("Failed to set env DANGER_GITLAB_HOST, error: %s", err)
+				}
+			}
+
+			if string(cfg.GitlabAPIBaseURL) != "" {
+				if err := os.Setenv("DANGER_GITLAB_API_BASE_URL", string(cfg.GitlabAPIBaseURL)); err != nil {
+					failf("Failed to set env DANGER_GITLAB_API_BASE_URL, error: %s", err)
+				}
+			}
+		}
+	}
+
+	//
+	// Check dependencies
+	log.Infof("Checking dependencies")
+	{
+		log.Printf("Bundler...")
+		if ok, err := bundlerInstalled(); err != nil {
+			failf("Failed to check bundler, error: %s", err)
+		} else if !ok {
+			log.Warnf(`Bundler is not installed`)
+			fmt.Println()
+			log.Printf("Install Bundler")
+
+			if cmds, err := installBundler(); err != nil {
+				failf("Failed to create Bundler install command, error: %s", err)
+			} else {
+				for _, cmd := range cmds {
+					if out, err := cmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+						if errorutil.IsExitStatusError(err) {
+							log.Warnf("%s failed: %s", out)
```

Comment:
missing argument: 2 %s, only one arg

Reply:
i do not see the point of warning the error then failing with the error, what about this:
```
	if err != nil {
		if errorutil.IsExitStatusError(err) {
			failf("%s failed: %s", cmd.PrintableCommandArgs(), out)
		}
		failf("%s failed: %s", cmd.PrintableCommandArgs(), err)
	}
```